Exception Handling/Task3/UserTaskService.cs

```diff
@@ -12,25 +13,49 @@ public UserTaskService(IUserDao userDao)
             _userDao = userDao;
         }
 
-        public int AddTaskForUser(int userId, UserTask task)
+        public string AddTaskForUser(int userId, UserTask task)
         {
-            if (userId < 0)
-                return -1;
+            IUser user;
+            IList<IUserTask> tasks;
 
-            var user = _userDao.GetUser(userId);
-            if (user == null)
-                return -2;
 
-            var tasks = user.Tasks;
-            foreach (var t in tasks)
+            try
             {
-                if (string.Equals(task.Description, t.Description, StringComparison.OrdinalIgnoreCase))
-                    return -3;
-            }
+                if (userId < 0)
```

Comment:
@IhorStetskyi extract validation to the method, please. [Single responsibility](https://en.wikipedia.org/wiki/Single-responsibility_principle) good explains why should do it.

Reply:
Additional CheckForStringEqualityIgnoringCase, CheckForException, TryToCreateTask methods created to meet Single responsibility principle